rust/src/delta_datafusion.rs

```diff
@@ -318,21 +340,15 @@ impl TableProvider for DeltaTable {
             DeltaTable::schema(self).unwrap(),
         )?);
 
-        // each delta table must register a specific object store, since paths are internally
-        // handled relative to the table root.
-        let object_store_url = self.storage.object_store_url();
-        let url: &Url = object_store_url.as_ref();
-        session.runtime_env.register_object_store(
-            url.scheme(),
-            url.host_str().unwrap_or_default(),
-            self.object_store(),
-        );
+        register_store(&self, session.runtime_env.clone());
 
         // TODO we group files together by their partition values. If the table is partitioned
         // and partitions are somewhat evenly distributed, probably not the worst choice ...
         // However we may want to do some additional balancing in case we are far off from the above.
         let mut file_groups: HashMap<Vec<ScalarValue>, Vec<PartitionedFile>> = HashMap::new();
-        if let Some(Some(predicate)) = (!filters.is_empty()).then_some(combine_filters(filters)) {
+        if let Some(Some(predicate)) =
+            (!filters.is_empty()).then_some(conjunction(filters.to_vec()))
```

Comment:
Perhaps:

```suggestion
        if let Some(predicate) = conjunction(filters)
```

([conjunction() returns None if input is empty](https://github.com/apache/arrow-datafusion/blob/7cba7582a67c9353dfd851243be048149c849c2c/datafusion/optimizer/src/utils.rs#L127))